internal/persistence/sql/uuid_mapping.go

```diff
@@ -24,17 +23,32 @@ func (UUIDMapping) TableName() string {
 	return "keto_uuid_mappings"
 }
 
-func (p *Persister) AddUUIDMapping(ctx context.Context, id uuid.UUID, representation string) error {
-	m := &UUIDMapping{
-		ID:                   id,
-		StringRepresentation: representation,
-	}
+func (p *Persister) ToUUID(ctx context.Context, text string) (uuid.UUID, error) {
+	id := uuid.NewV5(p.NetworkID(ctx), text)
 	p.d.Logger().Trace("adding UUID mapping")
 
-	return sqlcon.HandleError(p.Connection(ctx).Create(m))
+	// We need to write manual SQL here because the INSERT should not fail if
+	// the UUID already exists, but we still want to return an error if anything
+	// else goes wrong.
+	var query string
+	switch d := p.Connection(ctx).Dialect.Name(); d {
+	case "mysql":
+		query = `
+			INSERT IGNORE INTO keto_uuid_mappings (id, string_representation)
+			VALUES (?, ?)`
+	default:
+		query = `
+			INSERT INTO keto_uuid_mappings (id, string_representation)
+			VALUES (?, ?)
+			ON CONFLICT (id) DO NOTHING`
+	}
+
+	return id, sqlcon.HandleError(
+		p.Connection(ctx).RawQuery(query, id, text).Exec(),
+	)
```

Comment:
You should be able to check for this error also using `sqlcon.HandleError` and `errors.Is`: https://github.com/ory/x/blob/08ea77b2a961bd23c170ddb076b10686c71e93f1/sqlcon/error.go#L18-L24

Reply:
Ah, thanks!

Reply:
I just tried that, and one big difference between telling the db that it should ignore the conflict and ignoring the conflict in Go is that in the former case, the transaction is cancelled.

Reply:
I see, and we might want to run this as part of some outer transaction. Then I guess we have to keep that.